README.md

```diff
@@ -644,20 +645,34 @@ callback function as a parameter as well as `offRequest` and `offResponse`
 functions which will remove a callback function from the `onRequest` or 
 `onResponse` set if it exists. 
 
-The `onRequest(callback)` function takes a `callback` function that will pass the xhr `request`
-Object to that callback. These callbacks are called synchronously, in the order registered
-and act as pre-request hooks for modifying the xhr `request` Object prior to making a request.
+The `onRequest(callback)` function takes a `callback` function that will pass the xhr `options`
+Object to that callback. These callbacks are called synchronously, in the order registered 
+and act as pre-request hooks for modifying the xhr `options` Object prior to making a request.
 
 Example:
 ```javascript
-const playerRequestHook = (request) => {
-  const requestUrl = new URL(request.uri);
+const playerRequestHook = (options) => {
+  const requestUrl = new URL(options.uri);
   requestUrl.searchParams.set('foo', 'bar');
-  request.uri = requestUrl.href;
+  options.uri = requestUrl.href;
```

Comment:
Will need to update these samples to return the `options` object (or a new object) in the same pattern as `beforeRequest`.

It would make sense to document the behavior here - each `onRequest` hook should return an options object and each will receive the previous hook's options object.

Also, we should document that the `onResponse` hooks do not behave this way - the `request` object is passed to each by reference.

Reply:
Yeah I wasn't 100% sure that we wanted to document it that way as it wasn't clear whether we would remove the `options` return/chaining after `beforeRequest` is removed. I agree it makes sense if we keep it that way for `onRequest` hooks.

Reply:
Added `options` return to the `onRequest` examples. Documented `onResponse` passing parameters by reference as well.